src/book.py

```diff
@@ -834,6 +834,35 @@ def detect_exchange(self, file_path: Path) -> Optional[str]:
 
         return None
 
+    def get_price_from_csv(self):
+        print("hi")
+        for platform, operations_a in misc.group_by(
+            self.operations, "platform"
+        ).items():
+            for timestamp, operations_b in misc.group_by(
+                operations_a, "utc_time"
+            ).items():
+                if len(operations_b) > 1:
+                    buytr = selltr = None
+                    for operation in operations_b:
+                        if isinstance(operation, tr.Buy):
+                            buytr = operation
+                        elif isinstance(operation, tr.Sell):
+                            selltr = operation
```

Comment:
ah forgot that case. In case of multiple buy and sell i would just skip that pair, because there is not enough information on which buy and sell belongs to which pair.
```csv
2021-06-09 06:32:58,Spot,Small assets exchange BNB,BNB,9.9E-7,""
2021-06-09 06:32:58,Spot,Small assets exchange BNB,BNB,0.00000347,""
2021-06-09 06:32:58,Spot,Small assets exchange BNB,ATA,-0.00547803,""
2021-06-09 06:32:58,Spot,Small assets exchange BNB,FRONT,-0.00040334,""
2021-06-09 06:32:58,Spot,Small assets exchange BNB,BNB,0.00001395,""
2021-06-09 06:32:58,Spot,Small assets exchange BNB,LTC,-0.00000784,""
2021-06-09 13:36:59,Spot,Small assets exchange BNB,BNB,9.9E-7,""
2021-06-09 13:36:59,Spot,Small assets exchange BNB,FRONT,-0.00040621,""
```